game-core/src/main/java/games/strategy/triplea/delegate/battle/end/NoUnitsWithRolls.java

```diff
@@ -0,0 +1,69 @@
+package games.strategy.triplea.delegate.battle.end;
+
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.Matches;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import lombok.Builder;
+import lombok.NonNull;
+import org.triplea.java.collections.CollectionUtils;
+
+@Builder
+public class NoUnitsWithRolls {
+  private final boolean isAttacker;
+  private final boolean hasRetreatTerritories;
+  private @NonNull final Collection<Unit> attackingUnits;
+  private @NonNull final Collection<Unit> defendingUnits;
+  private @NonNull final Territory battleSite;
+
+  public Collection<Unit> check() {
+    if (isRetreatPossible() || attackingUnits.isEmpty() || defendingUnits.isEmpty()) {
+      return List.of();
+    }
+    final Collection<Unit> units;
```

Comment:
This seems a bit strange. Can we simply not eliminate isAttacker boolean and have the caller pass units/enemyUnits in different args when it wants it one way or the other?

e.g. when using this builder:

```
            .attackingUnits(isAttacker ? attackingUnits : defendingUnits)
            .defendingUnits(isAttacker ? defendingUnits : attackingUnits)
```

Then, the code in the builder can be simplified.

Reply:
`isAttacker` is still needed for the call to `Matches#unitIsSupporterOrHasCombatAbility`.  But I like the idea of flipping the units in the call to the builder instead of inside of `check`.